### PR TITLE
feat: add bayesian_update tool for exact posterior computation

### PR DESCRIPTION
**Pull request by geemus**
Implements #127.

Adds a deterministic `bayesian_update` tool so forecaster LLMs can hand off Bayesian math and receive exact posteriors instead of approximating in their heads.

### Changes

| File | Change |
|------|--------|
| `lib/tools.rb` | `BAYESIAN_UPDATE_TOOL` schema, `bayesian_update` method, dispatch case. Supports `likelihood_ratios` and `evidence_items` input modes. Clamps prior to `1e-9` to avoid NaN/0/1. |
| `lib/prompts.rb` | Tool usage guidance in `SUPERFORECASTER_SYSTEM_PROMPT` |
| `forecast.rb` | Pass both `CALCULATOR_TOOL` + `BAYESIAN_UPDATE_TOOL` to LLM calls |
| `revise_forecast.rb` | Same — was previously missing tools entirely |
| `test/test_tools.rb` | **New**: 19 tests covering both input modes, edge cases, dispatch routing |
| `test/run_tests.rb` | Added `require_relative` for the new test file |

### Verification
- `bundle exec ruby test/run_tests.rb` → **73 runs, 542 assertions, 0 failures**

### CODE REVIEW

File: lib/prompts.rb [hunk 1]
```diff
@@ -32,6 +32,7 @@
   - Assign precise, justified numerical likelihoods (e.g., 42%, 2.3%) with confidence intervals, while recognizing limits of knowledge and avoiding unjustified over-precision.
   - Put extra weight on status quo outcomes since the world usually changes slowly.
   - For each adjustment — to the base rate, for cognitive/source biases, or to confidence — explicitly state the direction, magnitude, supporting evidence, and reasoning.
+  - Use the `bayesian_update` tool to compute exact posterior probabilities when adjusting base rates with evidence likelihood ratios. It is more accurate than mental arithmetic. Call it with your prior and one or more likelihood ratios; the tool returns the exact posterior.
```
Comment (geemus) [inline lib/prompts.rb]:
```suggestion
  - Use the `bayesian_update` tool to compute exact posterior probabilities when adjusting base rates with evidence likelihood ratios. It is more accurate than mental arithmetic. Call it with your prior and one or more likelihood ratios; the tool returns the exact posterior plus shift magnitude and combined LR for sanity-checking. For calibration: a typical piece of strong evidence has LR 3–10; definitive near-certain proof might warrant LR up to 100. If you find yourself estimating LR > 100, you are likely overconfident.
```
